rca/home/models.py

```diff
@@ -174,6 +177,24 @@ def clean(self):
         if errors:
             raise ValidationError(errors)
 
+    def get_news_and_events(self):
+        cache_key = "latest_news_and_events"
+        news_and_events_data = cache.get(cache_key)
+        if not news_and_events_data:
+            news_and_events_data = content.pull_news_and_events()
+            cache.set(
+                cache_key, news_and_events_data, settings.API_CONTENT_CACHE_TIMEOUT
+            )
+        return news_and_events_data
+
+    def get_alumni_stories(self):
+        cache_key = "latest_alumni_stories"
+        stories_data = cache.get(cache_key)
+        if not stories_data:
+            stories_data = content.pull_alumni_stories()
+            cache.set(cache_key, stories_data, settings.API_CONTENT_CACHE_TIMEOUT)
```

Comment:
You should not set cache if code inside `pull_alumni_stories` failed. If I were you, I'd raise a custom exception in `pull_alumni_stories` or check `if stories_data is None` (and ensure that is returned from the method when failure occurs).

Reply:
I've improved the exception handling so this is ready for re-review, thanks.